.github/actions/run-python-test-set/action.yml

```diff
@@ -104,7 +104,7 @@ runs:
           exit 1
         fi
         if [[ "${{ inputs.run_in_parallel }}" == "true" ]]; then
-          EXTRA_PARAMS="-n4 $EXTRA_PARAMS"
+          EXTRA_PARAMS="-n$(nproc) $EXTRA_PARAMS"
```

Comment:
FYI nproc is typically too much because one test process spawns many processes under the hood, one pageserver (sometimes two), three safekeepers, etcd, one or more computes. One way to check would be to ssh to the runner and see what the cpu usage is. For 4 physical cores and 8 with hyper threads I used 4 test processes. Worth experimenting with different values